org.hl7.fhir.publisher.core/src/main/java/org/hl7/fhir/igtools/publisher/xig/XIGDatabaseBuilder.java

```diff
@@ -429,113 +474,115 @@ public void processResource(PackageContext context, Object clientContext, String
-            psqlR.setString(17, cr.getDateElement().primitiveValue());
-            psqlR.setString(18, cr.getName());
-            psqlR.setString(19, cr.getTitle());
-            psqlR.setBoolean(20, cr.getExperimental());
-            psqlR.setString(21, realm);
-            psqlR.setString(22, cr.getDescription());
-            psqlR.setString(23, cr.getPurpose());
-            psqlR.setString(24, cr.getCopyright());
-            psqlR.setString(25, cr.getCopyrightLabel()); 
-            psqlR.setString(26, j.asString("kind"));
-            psqlR.setString(27, j.asString("type"));        
-            psqlR.setString(28, j.asString("supplements"));        
-            psqlR.setString(29, j.asString("valueSet"));        
-            psqlR.setString(30, j.asString("content"));         
-            psqlR.setString(31, auth);
-            psqlR.setString(32, details);
-            psqlR.setString(33, ExtensionUtilities.readStringExtension(cr, ExtensionDefinitions.EXT_STANDARDS_STATUS));
-            psqlR.setString(34, ExtensionUtilities.readStringExtension(cr, ExtensionDefinitions.EXT_FMM_LEVEL));
-            psqlR.setString(35, ExtensionUtilities.readStringExtension(cr, ExtensionDefinitions.EXT_WORKGROUP));
-            psqlR.execute();
-
-            psqlC.setInt(1, resKey);
-            psqlC.setBytes(2, gzip(content));
-            psqlC.setBytes(3, gzip(new JsonParser().composeBytes(cr)));
-            psqlC.execute();
-
-            psqlRI.setInt(1, resKey);
-            psqlRI.setString(2, cr.getName());
-            psqlRI.setString(3, cr.getTitle());
-            psqlRI.setString(4, cr.getDescription());
-            psqlRI.setString(5, narrative);
-            psqlRI.execute();
-
-            //            if (cr instanceof StructureDefinition) {
-            //              dep = processStructureDefinition(resKey, (StructureDefinition) cr);
-            //              ext = processStructureDefinition2(resKey, (StructureDefinition) cr);
-            //            }
-          }       
+              String rid = r.hasId() ? r.getId() : id.replace(".json", "");
+              psqlR.setInt(1, resKey);
+              psqlR.setInt(2, smm.getKey());
+              psqlR.setString(3, type);
+              psqlR.setString(4, r.fhirType());
+              psqlR.setString(5, rid);
+              psqlR.setInt(6, hasVersion(context.getVersion(), "1.0"));
+              psqlR.setInt(7, hasVersion(context.getVersion(), "1.4"));
+              psqlR.setInt(8, hasVersion(context.getVersion(), "3.0"));
+              psqlR.setInt(9, hasVersion(context.getVersion(), "4.0"));
+              psqlR.setInt(10, hasVersion(context.getVersion(), "4.3"));
+              psqlR.setInt(11, hasVersion(context.getVersion(), "5.0"));
+              psqlR.setInt(12, hasVersion(context.getVersion(), "6.0"));
+              psqlR.setString(13, Utilities.pathURL(smm.getBase(), smm.getPath(cr.getUrl(), null, cr.fhirType(), cr.getIdBase())));
```

Comment:
At this point, the URL generated is a little odd. I get this from a trace:

https://profiles.ihe.net/PCC/ODH/https://build.fhir.org/ig/IHE/PCC.ODH/index.html
https://profiles.ihe.net/PCC/ODH/https://build.fhir.org/ig/IHE/PCC.ODH/StructureDefinition-odh-CombatZonePeriod.html

Are these just supposed to be unique keys, or something parsed later?

Reply:
wait what? I'll debug again